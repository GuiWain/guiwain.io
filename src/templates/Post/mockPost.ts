export const post = {
  title: 'Nullam in nisl eget tellus bibendum.',
  slug: '1',
  resume:
    'Nullam in nisl eget tellus bibendum. Nullam in nisl eget tellus bibendum.',
  content:
    "<math><mi//xlink:href='data:x,<script>alert(4)</script>'><h2>The three greatest things you learn from traveling</h2><p>Like all the great things on earth traveling teaches us by example. Here are some of the most precious lessons I’ve learned over the years of traveling.</p><figure class='image image-style-side'><img src='https://ckeditor.com/docs/ckeditor5/latest/assets/img/volcano.jpg' alt='A lone wanderer looking at Mount Bromo volcano in Indonesia.' srcset='https://ckeditor.com/docs/ckeditor5/latest/assets/img/volcano.jpg, https://ckeditor.com/docs/ckeditor5/latest/assets/img/volcano_2x.jpg 2x' sizes='100vw'><figcaption>Leaving your comfort zone might lead you to such beautiful sceneries like this one.</figcaption></figure><h3>Appreciation of diversity</h3><p>Getting used to an entirely different culture can be challenging. While it’s also nice to learn about cultures online or from books, nothing comes close to experiencing cultural diversity in person. You learn to appreciate each and every single one of the differences while you become more culturally fluid.</p><blockquote><p>The real voyage of discovery consists not in seeking new landscapes, but having new eyes.</p><p><strong>Marcel Proust</strong></p></blockquote><h3>Improvisation</h3><p>Life doesn't allow us to execute every single plan perfectly. This especially seems to be the case when you travel. You plan it down to every minute with a big checklist; but when it comes to executing it, something always comes up and you’re left with your improvising skills. You learn to adapt as you go. Here’s how my travel checklist looks now:</p><ul><li>buy the ticket</li><li>start your adventure</li></ul><figure class='image image-style-side'><img src='https://ckeditor.com/docs/ckeditor5/latest/assets/img/umbrellas.jpg' alt='Three Monks walking on ancient temple.' srcset='https://ckeditor.com/docs/ckeditor5/latest/assets/img/umbrellas.jpg, https://ckeditor.com/docs/ckeditor5/latest/assets/img/umbrellas_2x.jpg 2x' sizes='100vw'><figcaption>Leaving your comfort zone might lead you to such beautiful sceneries like this one.</figcaption></figure><h3>Confidence</h3><p>Going to a new place can be quite terrifying. While change and uncertainty makes us scared, traveling teaches us how ridiculous it is to be afraid of something before it happens. The moment you face your fear and see there was nothing to be afraid of, is the moment you discover bliss.</p>",
  date: 'Novembro 10, 2022',
  reading_time: 10,
  updateDate: 'dezembro 16, 2022'
}

export const post2 = {
  title:
    'Aut totam quasi ut libero distinctio quo praesentium quidem qui sint omnis quo tempora placeat et illum laborum!',
  slug: '2',
  resume:
    'Id soluta reprehenderit aut temporibus sunt et velit laboriosam. Quo earum asperiores quos delectus 33 molestiae nobis.',
  content:
    "<h2>Handheld emperor</h2><p>Nintendo, a Japanese electronics company, started as a <a href='https://en.wikipedia.org/wiki/Hanafuda'><i>hanafuda</i> cards</a> manufacturer in 1889. In the mid-1970s, they entered the early video games market and became famous for their home video and handheld game consoles. Nintendo introduced consoles like <strong>NES</strong>, <strong>SNES</strong>, and <strong>Wii</strong>. But the most revolutionary was for sure the <strong>Game Boy</strong>.</p><h2>A countdown of Nintendo handhelds</h2><figure class='image image-style-side'><img src='https://ckeditor.com/docs/ckeditor5/latest/assets/img/game_boy.jpg'><figcaption>The Nintendo 'flagship' - handheld Game Boy</figcaption></figure><ol><li><strong>Game &amp; Watch</strong> was Nintendo's first product offering out-of-home gaming. From 1980 to 1991, over a hundred games were released, gaining great popularity.</li><li>In 1989, &nbsp;the original <strong>Game Boy</strong> was released. The new amazing machine utilized a cartridge system, allowing the users to play a multitude of games of all kinds. This was a historical game-changer. &nbsp;</li><li>In 2004, Nintendo introduced a new console family called the <strong>Nintendo DS</strong>. It sported a dual LCD screen in a folded shell, with the lower one being a touchscreen.</li><li>2017 brought the hybrid experience for both couch-preferring gamers and handheld enthusiasts with the release of the <strong>Nintendo Switch</strong>. It offers both a TV mode with high-definition graphics and a handheld mode using the built-in 6.2” display.&nbsp;</li></ol><h2>Handheld consoles' popularity</h2><p>While the most recent Switch is a prevalent choice nowadays, the 2DS and 3DS consoles are still popular. The king, however, is none other than the original wonder — the Game Boy.</p><figure class='table'><table><tbody><tr><td>Console</td><td>Production dates</td><td>Pieces sold (2021)</td></tr><tr><td>Game &amp; Watch</td><td>1980-1991, 2020-2021</td><td>44 million</td></tr><tr><td>Game Boy</td><td>1989-2010</td><td>201 million 1</td></tr><tr><td>Nintendo DS</td><td>2011-2020</td><td>76 million 2</td></tr><tr><td>Switch</td><td>since 2017</td><td>93 million</td></tr><tr><td colspan='3'><p>1 119 million Game Boy and Game Boy Color variants, 82 million Game Boy Advance variants.</p><p>2 Including all versions: DS, DSi, 2DS, 3DS, and New 2DS/3DS variants.</p></td></tr></tbody></table></figure><h2>Handheld gaming experience</h2><blockquote><p>It's dangerous to go alone! Take this.&nbsp;<br>“Legend of Zelda”, 1986</p></blockquote><p>Games offered by Nintendo include multiple genres, out of which the famous platformer arcade <i>Super Mario</i> and the adventure roleplay <i>Legend of Zelda</i> series are probably the most iconic.</p><p>Games that can be played on the handheld family include (examples of games listed):</p><ul><li>Action &amp; adventure games<ul><li>The <i>Legend of Zelda</i> series</li><li><i>Chrono Trigger</i></li></ul></li><li>First-person action games<ul><li><i>Splatoon</i></li></ul></li><li>Role-playing games (RPG)<ul><li>The <i>Pokemon</i> series</li><li>The <i>Final Fantasy</i> series</li></ul></li></ul>",
  date: 'Setembro 22, 2022',
  reading_time: 4,
  updateDate: 'dezembro 16, 2022'
}

export const post3 = {
  title: 'Est fugit sequi eum molestiae molestiae et dolorum dolores.',
  slug: '3',
  resume:
    'Ea ratione iusto sed omnis sequi ut ipsam asperiores vel facere aperiam. Porro quos id placeat illum aut rerum voluptatem sed autem deleniti.',
  content:
    "<h2>Taj Mahal: A breathtaking ode to love</h2><figure class='image image-style-side'><img src='https://ckeditor.com/docs/ckeditor5/latest/assets/img/tajmahal.jpg' alt='Taj Mahal illustration.' srcset='https://ckeditor.com/docs/ckeditor5/latest/assets/img/tajmahal.jpg, https://ckeditor.com/docs/ckeditor5/latest/assets/img/tajmahal_2x.jpg 2x' sizes='100vw'><figcaption>Taj Mahal with its poetic white marble tomb</figcaption></figure><p>No trip to India is complete without visiting this spectacular monument, <a href='https://en.wikipedia.org/wiki/New7Wonders_of_the_World'><strong>counted among the Seven Wonders of the World</strong></a>.</p><p>Tourists frequently admit that Taj Mahal 'simply cannot be described with words'. And that’s probably true. The more you try the more speechless you become. Words give only a semblance of truth. The real truth about its beauty is revealed when you adore <strong>different shades of “Taj” depending on the time of the day</strong> or when you admire the exquisite inlay work in different corners of the façade.</p><h3>Masterpiece of the world’s heritage</h3><p>Taj Mahal is a mausoleum built in Agra between 1631 and 1648 by Emperor Shah Jahan <strong>in the memory of his beloved wife</strong>, Mumtaz Mahal, whose body lies there. It took 20 000 workers to complete and the excellence of this building is visible in every brick.</p><p>In 1983, Taj Mahal was appointed <a href='https://en.wikipedia.org/wiki/World_Heritage_Site'>UNESCO World Heritage Site</a> for being 'the jewel of Muslim art in India and one of the universally admired masterpieces of the world's heritage'.</p><p>If you like having a methodology for visiting historical places, here are the four elements on which we recommend to focus your attention:</p><ul><li>The tomb</li><li>The Decorations</li><li>The Garden</li><li>The Outlying buildings</li></ul><p>The tomb is what immediately catches your eyesight. The <strong>white and soft marble</strong> embroidered with stones leaves you totally enchanted.</p>",
  date: 'Novembro 10, 2022',
  reading_time: 10,
  updateDate: 'dezembro 16, 2022'
}
