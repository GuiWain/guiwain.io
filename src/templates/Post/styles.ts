import styled, { css } from 'styled-components'

import UseLink from 'next/link'

import { Calendar3 } from '@styled-icons/bootstrap/'
import { Update } from '@styled-icons/material/'
import { Timer } from '@styled-icons/boxicons-regular/'

export const Link = styled(UseLink)``

export const CalendarIcon = styled(Calendar3)`
  ${({ theme }) => css`
    width: 1rem;
    color: ${theme.colors.text};
  `}
`

export const UpdateDateIcon = styled(Update)`
  ${({ theme }) => css`
    width: 1.2rem;
    color: ${theme.colors.text};
  `}
`

export const TimerIcon = styled(Timer)`
  ${({ theme }) => css`
    width: 1.5rem;
    color: ${theme.colors.text};
  `}
`

export const Wrapper = styled.div`
  width: 90%;
  margin: 2rem auto 0;
`

export const Title = styled.h1`
  ${({ theme }) => css`
    color: ${theme.colors.title};
    font-family: ${theme.font.family01};
    font-size: ${theme.font.sizes.xlarge};
    font-weight: ${theme.font.extraBold};
  `}
`

export const Resume = styled.h2`
  ${({ theme }) => css`
    color: ${theme.colors.text};
    font-family: ${theme.font.family01};
    font-size: ${theme.font.sizes.xmedium};
    font-weight: ${theme.font.bold};

    margin: 1.5rem auto 0.5rem;
  `}
`

export const Info = styled.div`
  display: flex;
  justify-content: flex-start;
  flex-wrap: wrap;
  gap: 0.5rem 1.5rem;
`

export const Date = styled.time`
  ${({ theme }) => css`
    color: ${theme.colors.text};
    font-family: ${theme.font.family02};
    font-size: ${theme.font.sizes.xxxsmall};
    font-weight: ${theme.font.light};

    display: flex;
    align-items: center;
    gap: 0.3rem;
  `}
`

export const UpdateDate = styled(Date)`
  ${({ theme }) => css`
    font-style: italic;
    margin: 0.5rem auto;
  `}
`

export const ReadingTime = styled.span`
  ${({ theme }) => css`
    color: ${theme.colors.text};
    font-family: ${theme.font.family02};
    font-size: ${theme.font.sizes.xxxsmall};
    font-weight: ${theme.font.light};

    display: flex;
    align-items: center;
    gap: 0.3rem;
  `}
`

export const Content = styled.section`
  ${({ theme }) => css`
    margin-top: 5rem;

    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
      color: ${theme.colors.text};
      font-family: ${theme.font.family02};
      font-weight: ${theme.font.bold};

      margin: 2rem auto 1rem;
    }

    h1 {
      font-size: ${theme.font.sizes.xlarge};
    }

    h2 {
      font-size: ${theme.font.sizes.large};
    }

    h3 {
      font-size: ${theme.font.sizes.xmedium};
    }

    h4,
    h5,
    h6 {
      font-size: ${theme.font.sizes.medium};
    }

    p {
      margin: 0 auto 1.75rem;

      color: ${theme.colors.text};
      font-family: ${theme.font.family02};
      font-size: ${theme.font.sizes.xsmall};
      font-weight: ${theme.font.regular};

      word-wrap: break-word;
      line-height: 2.75rem;
      letter-spacing: 0.05rem;
    }

    a {
      color: ${theme.colors.title};
      font-family: ${theme.font.family02};
      font-size: ${theme.font.sizes.xsmall};
      font-weight: ${theme.font.regular};

      transition: text-decoration ${theme.transition.default};

      :hover {
        text-decoration: none;
      }
    }

    span {
      color: ${theme.colors.text};
      font-family: ${theme.font.family02};
      font-size: ${theme.font.sizes.xxsmall};
      font-weight: ${theme.font.regular};
    }

    li,
    ol,
    ul {
      color: ${theme.colors.text};
      font-family: ${theme.font.family02};
      font-size: ${theme.font.sizes.xxsmall};
      font-weight: ${theme.font.regular};

      line-height: 2.3rem;
      letter-spacing: 0.025rem;
    }

    ul,
    ol {
      width: 90%;
      margin: 1.5rem auto;
    }

    li {
      margin: 1.5rem auto;
    }

    figcaption {
      color: ${theme.colors.text};
      font-family: ${theme.font.family02};
      font-size: ${theme.font.sizes.little};
      font-weight: ${theme.font.light};

      margin-bottom: 1.5rem;
    }

    img {
      object-fit: contain;
      max-width: 100%;
    }

    td {
      color: ${theme.colors.text};
      font-family: ${theme.font.family02};
    }
  `}
`
