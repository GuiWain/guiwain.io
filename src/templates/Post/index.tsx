import * as S from './styles'

export type PostProps = {
  title: string
  resume?: string
  content: string
  date: string
  reading_time: number
  updateDate?: string
}

const Post = ({
  title,
  resume,
  content,
  date,
  reading_time,
  updateDate
}: PostProps) => {
  const createMarkup = (sanitizedContent: string) => {
    return { __html: sanitizedContent }
  }

  return (
    <S.Wrapper>
      <S.Info>
        <S.Date dateTime={date}>
          <S.CalendarIcon /> {date}
        </S.Date>

        <S.ReadingTime>
          <S.TimerIcon /> {reading_time} min. de leitura
        </S.ReadingTime>
      </S.Info>

      {!!updateDate && (
        <S.UpdateDate dateTime={date}>
          <S.UpdateDateIcon /> Atualizado em {date}
        </S.UpdateDate>
      )}

      <S.Title>{title}</S.Title>
      {!!resume && <S.Resume>{resume}</S.Resume>}

      <S.Content dangerouslySetInnerHTML={createMarkup(content)} />
    </S.Wrapper>
  )
}

export default Post
