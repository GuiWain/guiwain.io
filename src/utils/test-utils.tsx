import { ThemeProvider } from 'styled-components'
import { render, RenderResult } from '@testing-library/react'
import * as theme from 'styles/theme'

const renderWithTheme = (children: React.ReactNode): RenderResult =>
  render(<ThemeProvider theme={theme.light}>{children}</ThemeProvider>)

export * from '@testing-library/react'
export { renderWithTheme as render }
