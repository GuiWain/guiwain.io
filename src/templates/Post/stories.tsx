import { ComponentMeta, ComponentStory } from '@storybook/react'

import Post from '.'

export default {
  title: 'Components/Post',
  component: Post
} as ComponentMeta<typeof Post>

export const Default: ComponentStory<typeof Post> = () => <Post />
