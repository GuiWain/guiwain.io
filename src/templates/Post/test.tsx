import { render } from 'utils/test-utils'

import Post from '.'

describe('<Post />', () => {
  it('should render the Post', () => {
    const { container } = render(<Post />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
