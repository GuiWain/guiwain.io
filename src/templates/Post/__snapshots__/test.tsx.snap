// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Post /> should render the Post 1`] = `
.c3 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c4 {
  width: 1rem;
  color: #2D2529;
}

.c6 {
  width: 1.5rem;
  color: #2D2529;
}

.c0 {
  width: 90%;
  margin: 2rem auto 0;
}

.c7 {
  color: #7A0041;
  font-family: Montserrat,sans-serif;
  font-size: 2.8rem;
  font-weight: 800;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  gap: 0.5rem 1.5rem;
}

.c2 {
  color: #2D2529;
  font-family: Merriweather,serif;
  font-size: 1.2rem;
  font-weight: 300;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.3rem;
}

.c5 {
  color: #2D2529;
  font-family: Merriweather,serif;
  font-size: 1.2rem;
  font-weight: 300;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.3rem;
}

.c8 {
  margin-top: 5rem;
}

.c8 h1,
.c8 h2,
.c8 h3,
.c8 h4,
.c8 h5,
.c8 h6 {
  color: #2D2529;
  font-family: Merriweather,serif;
  font-weight: 700;
  margin: 2rem auto 1rem;
}

.c8 h1 {
  font-size: 2.8rem;
}

.c8 h2 {
  font-size: 2.4rem;
}

.c8 h3 {
  font-size: 2.2rem;
}

.c8 h4,
.c8 h5,
.c8 h6 {
  font-size: 2.0rem;
}

.c8 p {
  margin: 0 auto 1.75rem;
  color: #2D2529;
  font-family: Merriweather,serif;
  font-size: 1.6rem;
  font-weight: 400;
  word-wrap: break-word;
  line-height: 2.75rem;
  -webkit-letter-spacing: 0.05rem;
  -moz-letter-spacing: 0.05rem;
  -ms-letter-spacing: 0.05rem;
  letter-spacing: 0.05rem;
}

.c8 a {
  color: #7A0041;
  font-family: Merriweather,serif;
  font-size: 1.6rem;
  font-weight: 400;
  -webkit-transition: text-decoration 0.4s ease-in-out;
  transition: text-decoration 0.4s ease-in-out;
}

.c8 a:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c8 span {
  color: #2D2529;
  font-family: Merriweather,serif;
  font-size: 1.4rem;
  font-weight: 400;
}

.c8 li,
.c8 ol,
.c8 ul {
  color: #2D2529;
  font-family: Merriweather,serif;
  font-size: 1.4rem;
  font-weight: 400;
  line-height: 2.3rem;
  -webkit-letter-spacing: 0.025rem;
  -moz-letter-spacing: 0.025rem;
  -ms-letter-spacing: 0.025rem;
  letter-spacing: 0.025rem;
}

.c8 ul,
.c8 ol {
  width: 90%;
  margin: 1.5rem auto;
}

.c8 li {
  margin: 1.5rem auto;
}

.c8 figcaption {
  color: #2D2529;
  font-family: Merriweather,serif;
  font-size: 1.0rem;
  font-weight: 300;
  margin-bottom: 1.5rem;
}

.c8 img {
  object-fit: contain;
  max-width: 100%;
}

.c8 td {
  color: #2D2529;
  font-family: Merriweather,serif;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <time
      class="c2"
    >
      <svg
        aria-hidden="true"
        class="c3 c4"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 16 16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14 0H2a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2zM1 3.857C1 3.384 1.448 3 2 3h12c.552 0 1 .384 1 .857v10.286c0 .473-.448.857-1 .857H2c-.552 0-1-.384-1-.857V3.857z"
        />
        <path
          d="M6.5 7a1 1 0 1 0 0-2 1 1 0 0 0 0 2zm3 0a1 1 0 1 0 0-2 1 1 0 0 0 0 2zm3 0a1 1 0 1 0 0-2 1 1 0 0 0 0 2zm-9 3a1 1 0 1 0 0-2 1 1 0 0 0 0 2zm3 0a1 1 0 1 0 0-2 1 1 0 0 0 0 2zm3 0a1 1 0 1 0 0-2 1 1 0 0 0 0 2zm3 0a1 1 0 1 0 0-2 1 1 0 0 0 0 2zm-9 3a1 1 0 1 0 0-2 1 1 0 0 0 0 2zm3 0a1 1 0 1 0 0-2 1 1 0 0 0 0 2zm3 0a1 1 0 1 0 0-2 1 1 0 0 0 0 2z"
        />
      </svg>
       
    </time>
    <span
      class="c5"
    >
      <svg
        aria-hidden="true"
        class="c3 c6"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m20.145 8.27 1.563-1.563-1.414-1.414L18.586 7c-1.05-.63-2.274-1-3.586-1-3.859 0-7 3.14-7 7s3.141 7 7 7 7-3.14 7-7a6.966 6.966 0 0 0-1.855-4.73zM15 18c-2.757 0-5-2.243-5-5s2.243-5 5-5 5 2.243 5 5-2.243 5-5 5z"
        />
        <path
          d="M14 10h2v4h-2zm-1-7h4v2h-4zM3 8h4v2H3zm0 8h4v2H3zm-1-4h3.99v2H2z"
        />
      </svg>
       
       min. de leitura
    </span>
  </div>
  <h1
    class="c7"
  />
  <section
    class="c8"
  />
</div>
`;
